File: 240913/최소공배수 구하기/find-the-least-common-multiple.cpp
#include <iostream>
using namespace std;

int gcd(int a, int b) {
    while (b != 0 ) {
        int r = a % b;
        a = b;
        b = r;
    }
    return a;
}

int larger(int a, int b) {
    if (a > b) {
        return a*b / gcd(a, b);
    }
    else if (a < b) {
        return a*b / gcd(b, a);
    }
    else {
        return a;
    }
}

int main() {
    int n, m;
    cin >> n >> m;
    cout << larger(n, m) <<"\n";
    return 0;
}

File: 240913/최소공배수 구하기/find-the-least-common-multiple.py
def gcd(a, b):
    while b != 0:
        r = a % b
        a = b
        b = r
    
    return a

def larger(a, b):
    if a > b:
        return a*b // gcd(a, b)
    elif a < b:
        return a*b // gcd(b, a)
    else:
        return a

n, m = map(int, input().split())
print(larger(n, m))